charts/dapr/charts/dapr_sidecar_injector/templates/dapr_sidecar_injector_deployment.yaml

```diff
@@ -154,6 +154,11 @@ spec:
 {{- if not .Values.global.actors.enabled }}
         - name: SKIP_PLACEMENT
           value: "true"
+{{- else }}
+{{- if .Values.global.ha.enabled }}
+        - name: PLACEMENT_ADDRESSES
+          value: "dapr-placement-server-0.dapr-placement-server.{{ .Release.Namespace }}.svc.cluster.local:50005,dapr-placement-server-1.dapr-placement-server.{{ .Release.Namespace }}.svc.cluster.local:50005,dapr-placement-server-2.dapr-placement-server.{{ .Release.Namespace }}.svc.cluster.local:50005"
```

Comment:
This is important for the sidecar to pick the endpoint instead of relying on the k8s service to pick which one to route to.

Reply:
> instead of relying on the k8s service to pick which one to route to.
Could you elaborate on this a bit? Placement is using a [headless](https://github.com/dapr/dapr/blob/b99cf8b5476a503f655d74ce478e681a1f9a37f0/charts/dapr/charts/dapr_placement/templates/dapr_placement_service.yaml#L26) service which will return the same thing but will be more correct, i.e. correct port, removes unhealthy hosts.

Reply:
How does this value interact with the Placement host address annotation that can be set in the app?

Reply:
Responding to Josh. Kubernetes has no idea about which one is the leader, so the sidecars have to keep reconnecting to the service until one of them accepts as the leader. Only the leader node can accept the streaming connections and do placement table disseminations. Sidecar already has logic to try multiple addresses until it finds the leader.

Reply:
> How does this value interact with the Placement host address annotation that can be set in the app?

It does not make a difference because I changed the default value, way before the annotation is handled.

Reply:
Indeed however daprd currently handles this by DNS querying the single headless service, then finds the leader through round robin on the returned set.

https://github.com/dapr/dapr/blob/b99cf8b5476a503f655d74ce478e681a1f9a37f0/pkg/actors/placement/placement.go#L129
https://github.com/dapr/dapr/blob/b99cf8b5476a503f655d74ce478e681a1f9a37f0/pkg/actors/placement/placement.go#L596C6-L596C26
https://github.com/dapr/dapr/blob/b99cf8b5476a503f655d74ce478e681a1f9a37f0/pkg/actors/placement/placement.go#L137
https://github.com/dapr/dapr/blob/b99cf8b5476a503f655d74ce478e681a1f9a37f0/pkg/actors/placement/grpc_opts.go#L65-L69

Reply:
Great. I will remove this from the sidecar injector. Learned something new. Thanks.

Reply:
Resolving this comment since I deleted this from the solution.